src/XRP/model/xrp-account-set.ts

```diff
@@ -16,15 +17,56 @@ export default class XrpAccountSet {
    * @return an XrpAccountSet with its fields set via the analogous protobuf fields.
    * @see https://github.com/ripple/rippled/blob/3d86b49dae8173344b39deb75e53170a9b6c5284/src/ripple/proto/org/xrpl/rpc/v1/transaction.proto#L100
    */
-  public static from(accountSet: AccountSet): XrpAccountSet | undefined {
+  public static from(accountSet: AccountSet): XrpAccountSet {
     const clearFlag = accountSet.getClearFlag()?.getValue()
     const domain = accountSet.getDomain()?.getValue()
+    // domain must be lowercase
+    if (domain?.toLowerCase() != domain) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'AccountSet protobuf field `domain` is not lowercase.',
+      )
+    }
     const emailHash = accountSet.getEmailHash()?.getValue_asU8()
     const messageKey = accountSet.getMessageKey()?.getValue_asU8()
     const setFlag = accountSet.getSetFlag()?.getValue()
+    if (clearFlag && setFlag && clearFlag == setFlag) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'AccountSet protobuf fields `clearFlag` and `setFlag` are equal.',
+      )
+    }
     const transferRate = accountSet.getTransferRate()?.getValue()
+    // transferRate cannot be more than 2000000000 or less than 1000000000,
+    // except for the special case 0 meaning no fee.
+    const maxTransferRate = 2000000000
+    const minTransferRate = 1000000000
+    const specialCaseTransferRate = 0
```

Comment:
you can move these inside of the if below, since they're only relevant in that block